<a href="https://colab.research.google.com/github/JFFBMA/Data-Analysis/blob/main/Pr%C3%A1cticas/DA_Sprint03_Teorico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Sprint 3 — Sesión teórica — Versión estudiante
## Introducción a SQL y KPIs financieros en Google Colab con DuckDB y Parquet

> **Duración:** 70 minutos  
> **Herramienta principal:** Google Colab  
> **Motor SQL:** DuckDB en memoria  
> **Dataset:** tablas Parquet cargadas desde GitHub  
> **Prefijo de archivos:** `webinar03_`  
> **Caso:** librería online / bookstore finance

Esta sesión consolida las sesiones teóricas del Sprint 3. El flujo original usaba una base local y archivos Parquet en una carpeta del computador. En esta versión, las tablas se leen desde GitHub y se crean dentro de Google Colab para que todos trabajen con el mismo entorno.

URL base esperada para los archivos:

```text
https://github.com/ljpiere/tpdata_course/raw/refs/heads/main/DA/DA_tutor/datasets/webinar03_*.parquet
```

## Cómo usar esta versión estudiante

Este notebook está preparado para que tomes apuntes, completes ejercicios y documentes tus decisiones durante la clase.

**Modo de trabajo recomendado:**

1. Lee primero la consigna de cada bloque o ejercicio.
2. Completa los espacios de respuesta antes de comparar con la explicación del instructor/a.
3. Ejecuta las celdas de setup cuando existan.
4. Escribe interpretación, dudas y decisiones en los espacios de apuntes.
5. Al final, revisa si tus respuestas conectan datos, método e implicación de negocio.

> Las salidas de ejecución fueron limpiadas para que puedas correr el notebook desde cero.


## Notas de clase principales

- SQL permite consultar datos tabulares usando instrucciones declarativas como `SELECT`, `WHERE`, `JOIN` y `GROUP BY`.
- DuckDB en Colab permite trabajar con archivos Parquet como si fueran tablas SQL.
- Antes de calcular KPIs, se validan cargas, conteos y relaciones entre tablas.
- Los `JOIN` agregan contexto, pero un join incorrecto puede duplicar filas e inflar métricas.
- Los KPIs financieros requieren distinguir revenue, costo, utilidad bruta, margen y ROI.

### Mis notas iniciales

-ya tengo nociones de SQL y de las métricas que vamos a trabajar
-
-


## Agenda de la sesión

| Momento | Duración | Propósito |
|---|---:|---|
| 1. SQL en contexto | 8 min | Entender qué es SQL y por qué se usa en analítica |
| 2. Entorno Colab + DuckDB | 10 min | Crear tablas SQL desde Parquet en GitHub |
| 3. Modelo relacional Bookstore | 8 min | Reconocer tablas, llaves y relaciones |
| 4. Modelo mental de una consulta SQL | 8 min | Entender `FROM`, `WHERE`, `GROUP BY`, `SELECT`, `ORDER BY` |
| 5. Consultas básicas | 10 min | Usar `SELECT`, `WHERE`, `ORDER BY`, `LIMIT` |
| 6. Agregaciones y joins | 12 min | Construir KPIs con `GROUP BY` y `JOIN` |
| 7. KPIs financieros | 10 min | Calcular revenue, cost, gross profit, margin y ROI |
| 8. Cierre | 4 min | Validar aprendizaje, takeaways y canales de ayuda |

## Objetivos de aprendizaje

Al finalizar la sesión, el estudiante podrá:

- Explicar qué es SQL y cómo se usa en analítica de datos.
- Diferenciar entre tabla, archivo Parquet, motor SQL y notebook.
- Cargar tablas desde GitHub en Google Colab.
- Consultar datos usando DuckDB.
- Validar conteos antes de analizar.
- Usar `SELECT`, `WHERE`, `ORDER BY`, `LIMIT`, `GROUP BY` y `JOIN`.
- Construir KPIs financieros básicos: revenue, cost, gross profit, margin % y ROI.
- Traducir un resultado SQL a una interpretación de negocio.

## 1) SQL en contexto: qué es y para qué sirve

SQL significa **Structured Query Language**. Es un lenguaje declarativo para consultar, transformar y resumir datos organizados en tablas.

La palabra clave es **declarativo**: no le decimos al motor cada paso físico. Le decimos qué resultado queremos obtener. El motor decide cómo leer, filtrar, unir y calcular internamente.

En analítica de datos, SQL sirve para responder preguntas como:

- ¿Cuántas órdenes se hicieron?
- ¿Qué categoría generó más revenue?
- ¿Qué segmento de clientes paga más?
- ¿Qué productos tienen mayor margen?
- ¿Qué campañas tienen mejor ROI?

Flujo de trabajo de esta clase:

```text
Parquet en GitHub → Google Colab → DuckDB → SQL → Resultado tabular → Interpretación de negocio

Parquet es un archivo liviano que contiene datos

```

### Mis respuestas de validación

1.  Parquet
2.  Github
3.  DuckDB

**Tema que necesito repasar:**  


## 2) Setup técnico en Google Colab

Ejecuta la siguiente celda al inicio de la clase.

La celda hace cuatro cosas:

1. instala dependencias;
2. define la URL base del repositorio;
3. lee cada archivo `.parquet`;
4. crea tablas SQL dentro de DuckDB.

Las tablas creadas serán:

- `customers`
- `products`
- `orders`
- `order_items`
- `payments`
- `campaigns`

### Mis apuntes

-Instalar dependencias no lo sabía
-
-


In [ ]:
# Instalaremos un motor de base de datos llamado duckdb
%pip install duckdb pandas pyarrow

In [ ]:
# Setup de Google Colab para practicar SQL con DuckDB y Parquet desde GitHub

import duckdb
import pandas as pd

BASE_URL = "https://raw.githubusercontent.com/ljpiere/tpdata_python/refs/heads/main/DA/DA_students/datasets/"

TABLE_FILES = {
    "customers": "sprint03_customers.parquet",
    "products": "sprint03_products.parquet",
    "orders": "sprint03_orders.parquet",
    "order_items": "sprint03_order_items.parquet",
    "payments": "sprint03_payments.parquet",
    "campaigns": "sprint03_campaigns.parquet"
}

con = duckdb.connect(database=":memory:")

for table_name, file_name in TABLE_FILES.items():
    url = f"{BASE_URL}/{file_name}"
    df = pd.read_parquet(url)
    con.register(f"{table_name}_df", df)
    con.execute(f"CREATE OR REPLACE TABLE {table_name} AS SELECT * FROM {table_name}_df")

def sql(query):
    """Ejecuta una consulta SQL en DuckDB y devuelve un DataFrame de pandas."""
    return con.execute(query).df()

print("Tablas cargadas en DuckDB:")
print(list(TABLE_FILES.keys()))


Tablas cargadas en DuckDB:
['customers', 'products', 'orders', 'order_items', 'payments', 'campaigns']


## 3) Verificación de carga

Antes de analizar, validamos que las tablas existan y que tengan filas.

Resultado esperado:

| tabla | filas |
|---|---:|
| customers | 200 |
| products | 80 |
| orders | 600 |
| order_items | 902 |
| payments | 600 |
| campaigns | 6 |

Esta validación evita analizar datos incompletos o mal cargados.

### Mis respuestas de validación

1.  
2.  
3.  

**Tema que necesito repasar:**  


In [ ]:
sql("""
SELECT 'customers' AS tabla, COUNT(*) AS filas FROM customers
UNION ALL
SELECT 'products', COUNT(*) FROM products
UNION ALL
SELECT 'orders', COUNT(*) FROM orders
UNION ALL
SELECT 'order_items', COUNT(*) FROM order_items
UNION ALL
SELECT 'payments', COUNT(*) FROM payments
UNION ALL
SELECT 'campaigns', COUNT(*) FROM campaigns;
""")

,tabla,filas
0,customers,200
1,products,80
2,orders,600
3,order_items,902
4,payments,600
5,campaigns,6


## 4) Modelo relacional del dataset

El dataset representa una librería online.

| Tabla | Qué contiene | Llave principal o de unión |
|---|---|---|
| `customers` | Clientes | `customer_id` |
| `products` | Libros/productos | `product_id` |
| `orders` | Órdenes | `order_id`, `customer_id`, `promo_code` |
| `order_items` | Líneas de cada orden | `order_id`, `product_id` |
| `payments` | Pagos por orden | `order_id` |
| `campaigns` | Campañas promocionales | `promo_code` |

Modelo mental:

```text
customers ── customer_id ── orders ── order_id ── payments
                                  │
                                  ├── order_items ── product_id ── products
                                  │
                                  └── promo_code ── campaigns
```

Una consulta analítica normalmente necesita responder tres preguntas:

| Pregunta | Ejemplo |
|---|---|
| ¿Qué tabla tiene la métrica? | `payments.amount_paid`, `order_items.quantity` |
| ¿Qué tabla tiene el contexto? | `products.category`, `customers.segment` |
| ¿Qué llave permite unirlas? | `order_id`, `product_id`, `customer_id` |

### Mis respuestas de validación

1.  
2.  
3.  

**Tema que necesito repasar:**  


## 5) Modelo mental de una consulta SQL

Aunque escribimos una consulta empezando por `SELECT`, el orden lógico es:

```text
FROM      → de dónde salen los datos
WHERE     → qué filas se conservan
GROUP BY  → cómo se agrupan las filas
HAVING    → qué grupos se conservan
SELECT    → qué columnas o cálculos se muestran
ORDER BY  → cómo se ordena el resultado
LIMIT     → cuántas filas se muestran
```

Ejemplo:

```sql
SELECT
    status,
    COUNT(*) AS n_orders
FROM orders
WHERE order_date >= '2025-01-01'
GROUP BY status
ORDER BY n_orders DESC;
```

Lectura humana: toma la tabla `orders`, conserva órdenes desde 2025, agrupa por estado, cuenta órdenes y ordena el resultado de mayor a menor.

### Mis apuntes
 repasar el having
-
-


In [ ]:
sql("""
SELECT
    status,
    COUNT(*) AS n_orders
FROM orders
WHERE order_date >= '2025-01-01'
GROUP BY status
ORDER BY n_orders DESC;
""")

,status,n_orders
0,completed,469
1,pending,53
2,cancelled,48
3,refunded,30


## 6) Consultas básicas: `SELECT`, `WHERE`, `ORDER BY`, `LIMIT`

Primero exploramos una tabla sin unirla con otras.

Puntos clave:

- `SELECT *` sirve para explorar, no para reportes finales.
- `WHERE` filtra filas.
- `ORDER BY` ordena.
- `LIMIT` controla cuántas filas vemos.

### Mis apuntes

- customers : customer_id	customer_name	email	country	city	segment	signup_date

- products:product_id	title	author	category	price	cost	is_active

- orders:order_id	customer_id	order_date	status	sales_channel	promo_code

- order_items:order_item_id	order_id	product_id	quantity	unit_price	discount_rate

- payments:payment_id	order_id	payment_method	gross	discount	tax	shipping_fee	amount_paid	currency

- campaigns:promo_code	campaign_name	channel	start_date	end_date	spend_usd




In [ ]:
sql("""
select * from campaigns
""")

,promo_code,campaign_name,channel,start_date,end_date,spend_usd
0,PROMO10,Lecturas de enero,email,2025-01-01,2025-01-31,1200.0
1,DATA15,Semana Data,paid_search,2025-02-10,2025-03-05,1800.0
2,BUS20,Impulso Business,social,2025-03-15,2025-04-15,1600.0
3,TECH10,Tech Readers,display,2025-04-01,2025-05-10,1400.0
4,READ5,Reactiva lectores,email,2025-05-01,2025-05-31,750.0
5,JUN15,Cierre semestre,affiliate,2025-06-01,2025-06-30,1300.0


## 7) Agregaciones: pasar de filas a KPIs

Un KPI resume muchas filas en una métrica.

Ejemplo: número de órdenes por estado.

### Espacio de trabajo del estudiante

**Mi interpretación de la consigna:**  

**Mi procedimiento / fórmula / consulta:**  

**Resultado obtenido:**  

**Conclusión breve:**  


In [ ]:
# TODO estudiante: completa la consulta SQL de acuerdo con la consigna anterior.
# Sugerencia:
# 1. Identifica la tabla base.
# 2. Define la métrica y la dimensión.
# 3. Valida si necesitas JOIN, WHERE, GROUP BY u ORDER BY.

query = """
select status, count(*) as n_orders
from orders
group by status
order by n_orders desc;
"""

sql(query)


,status,n_orders
0,completed,469
1,pending,53
2,cancelled,48
3,refunded,30


## 8) Primer `JOIN`: unir contexto de cliente, orden y pago

Un `JOIN` no es solo “pegar tablas”. Es declarar una relación entre entidades usando llaves.

En este ejemplo:

- `orders` tiene la orden;
- `customers` agrega el segmento;
- `payments` agrega el valor pagado.

### Mis apuntes

-
-
-


In [ ]:
sql("""
select
  *
from
  orders
join customers
  on orders.customer_id = customers.customer_id
join payments
  on orders.order_id = payments.order_id
""")

,order_id,customer_id,order_date,status,sales_channel,promo_code,customer_id_1,customer_name,email,country,...,signup_date,payment_id,order_id_1,payment_method,gross,discount,tax,shipping_fee,amount_paid,currency
0,1001,29,2025-06-17,completed,web,None,29,David Ruiz,customer029@example.com,Colombia,...,2024-05-05,PAY-1001,1001,credit_card,93.30,0.00,7.46,4.99,105.75,USD
1,1002,185,2025-06-15,completed,web,None,185,Laura Rojas,customer185@example.com,Colombia,...,2024-03-22,PAY-1002,1002,bank_transfer,42.55,0.00,3.40,0.00,45.95,USD
2,1003,34,2025-04-19,completed,marketplace,None,34,Daniela Torres,customer034@example.com,Colombia,...,2024-05-14,PAY-1003,1003,credit_card,118.00,0.27,9.42,0.00,127.15,USD
3,1004,57,2025-01-29,cancelled,call_center,PROMO10,57,Daniela Gómez,customer057@example.com,Colombia,...,2024-05-06,PAY-1004,1004,credit_card,84.75,0.00,0.00,0.00,0.00,USD
4,1005,31,2025-04-15,completed,web,None,31,Sebastián Mejía,customer031@example.com,Colombia,...,2024-02-10,PAY-1005,1005,credit_card,38.83,0.00,3.11,0.00,41.94,USD
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
595,1596,76,2025-04-27,refunded,web,None,76,Isabella Pérez,customer076@example.com,Colombia,...,2024-03-04,PAY-1596,1596,wallet,26.55,0.00,2.12,0.00,0.00,USD
596,1597,104,2025-03-16,completed,web,BUS20,104,Sebastián Martínez,customer104@example.com,Colombia,...,2023-01-07,PAY-1597,1597,credit_card,22.06,4.19,1.43,2.99,22.29,USD
597,1598,149,2025-01-17,completed,mobile_app,PROMO10,149,Valentina Castro,customer149@example.com,Colombia,...,2025-01-09,PAY-1598,1598,pse,68.06,7.22,4.87,4.99,70.70,USD
598,1599,147,2025-04-29,completed,call_center,None,147,Ana Pérez,customer147@example.com,Colombia,...,2023-02-01,PAY-1599,1599,credit_card,61.78,0.30,4.92,2.99,69.39,USD


## 9) KPI 1 — Revenue por categoría

Pregunta de negocio:

> ¿Qué categorías generan más revenue?

Fórmula:

```text
revenue = quantity * unit_price * (1 - discount_rate)
```

El revenue se calcula desde `order_items` porque allí están cantidad, precio unitario y descuento por línea. La categoría está en `products`, por eso necesitamos un `JOIN`.

### Espacio de trabajo del estudiante

**Mi interpretación de la consigna:**  

**Mi procedimiento / fórmula / consulta:

SUM(oi.quantity *  oi.unit_price * (1- oi.discount_rate)) as revenue**  

**Resultado obtenido:
category	revenue

0	Technology	10828.4329

1	Design	5807.8009

2	Marketing	5362.4994

3	Data	4573.5391

4	Business	3491.3563

5	Fiction	2321.8510

6	Productivity	1865.9182

7	Education	1307.943**  

**Conclusión breve:

Las tres primeras categorías son el 61.86% del revenue total, liderado por Technology**  


In [ ]:
# TODO estudiante: completa la consulta SQL de acuerdo con la consigna anterior.
# Sugerencia:
# 1. Identifica la tabla base.
# 2. Define la métrica y la dimensión.
# 3. Valida si necesitas JOIN, WHERE, GROUP BY u ORDER BY.

query = """
select
    p.category,
    SUM(oi.quantity *  oi.unit_price * (1- oi.discount_rate)) as revenue
from order_items oi
join products p
    on oi.product_id = p.product_id
group by
    p.category
order by
    revenue DESC;
"""
sql(query)


,category,revenue
0,Technology,10828.4329
1,Design,5807.8009
2,Marketing,5362.4994
3,Data,4573.5391
4,Business,3491.3563
5,Fiction,2321.8510
6,Productivity,1865.9182
7,Education,1307.9437


## 10) KPI 2 — Cost, gross profit y margin %

Pregunta de negocio:

> ¿Qué categorías no solo venden más, sino que dejan mejor rentabilidad?

Fórmulas:

```text
cost = quantity * product_cost
gross_profit = revenue - cost
margin_pct = 100 * gross_profit / revenue
```

Regla técnica: usamos `NULLIF(revenue, 0)` para evitar división por cero.

### Espacio de trabajo del estudiante

**Mi interpretación de la consigna:**  

**Mi procedimiento / fórmula / consulta:

REVENUE = SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate))

COST = SUM(oi.quantity * p.cost)

gross_profit = SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate))
     -
SUM(oi.quantity * p.cost)
    
margin_pct = 100 * (SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate))
     -
SUM(oi.quantity * p.cost)  )
    /
NULLIF(SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate)),0)
    

**Resultado obtenido:**
category	revenue	cost	gross_profit	margin_pct
0	Education	1307.9437	746.54	561.4037	42.922620
1	Technology	10828.4329	6345.79	4482.6429	41.396968
2	Data	4573.5391	2797.88	1775.6591	38.824618
3	Business	3491.3563	2141.38	1349.9763	38.666243
4	Fiction	2321.8510	1432.11	889.7410	38.320331
5	Productivity	1865.9182	1158.58	707.3382	37.908318
6	Design	5807.8009	3607.91	2199.8909	37.878208
7	Marketing	5362.4994	3374.14	1988.3594	37.07896

**Conclusión breve:**  EDUCACIÓN Y TECNOLOGÍA son las categorías con mayor margen o eficiencia operacional


In [ ]:
# TODO estudiante: completa la consulta SQL de acuerdo con la consigna anterior.
# Sugerencia:
# 1. Identifica la tabla base.
# 2. Define la métrica y la dimensión.
# 3. Valida si necesitas JOIN, WHERE, GROUP BY u ORDER BY.

query = """
SELECT
    p.category,
    SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate)) as revenue,
    SUM(oi.quantity * p.cost)  as cost,

    SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate))
     -
    SUM(oi.quantity * p.cost) as gross_profit,

    100 * (SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate))
     -
    SUM(oi.quantity * p.cost)  )
    /
    NULLIF(SUM(oi.quantity * oi.unit_price * (1- oi.discount_rate)),0)
    as margin_pct

FROM
    order_items oi
JOIN
    products p
    on oi.product_id = p.product_id
GROUP BY
    p.category
ORDER BY
    margin_pct DESC;
"""

sql(query)


,category,revenue,cost,gross_profit,margin_pct
0,Education,1307.9437,746.54,561.4037,42.922620
1,Technology,10828.4329,6345.79,4482.6429,41.396968
2,Data,4573.5391,2797.88,1775.6591,38.824618
3,Business,3491.3563,2141.38,1349.9763,38.666243
4,Fiction,2321.8510,1432.11,889.7410,38.320331
5,Productivity,1865.9182,1158.58,707.3382,37.908318
6,Design,5807.8009,3607.91,2199.8909,37.878208
7,Marketing,5362.4994,3374.14,1988.3594,37.078967


In [ ]:
product_id	title	author	category	price	cost	is_active

In [ ]:
order_item_id	order_id	product_id	quantity	unit_price	discount_rate

## 11) KPI 3 — ROI por campaña

Pregunta de negocio:

> ¿Qué campañas promocionales devuelven más valor respecto a su inversión?

Fórmula:

```text
ROI % = 100 * (net_revenue - spend_usd) / spend_usd
```

Supuesto de clase: `amount_paid` representa revenue neto cobrado a nivel de orden. Para KPIs operativos, excluimos órdenes `cancelled` y `refunded`.

### Espacio de trabajo del estudiante

**Mi interpretación de la consigna:**  

**Mi procedimiento / fórmula / consulta:**
SUM(pa.amount_paid) as ingreso_neto,
  SUM(ca.spend_usd) as inversion,
  SUM(pa.amount_paid) - SUM(ca.spend_usd) as Ganancia,
  100 * (SUM(pa.amount_paid) - SUM(ca.spend_usd)) / NULLIF(SUM(ca.spend_usd),0) as ROI

**Resultado obtenido:**  
0	Reactiva lectores	1222.14	13500.0	-12277.86	-90.947111
1	Cierre semestre	2085.65	37700.0	-35614.35	-94.467772
2	Lecturas de enero	1721.26	34800.0	-33078.74	-95.053851
3	Tech Readers	1244.88	30800.0	-29555.12	-95.958182
4	Impulso Business	1428.60	38400.0	-36971.40	-96.279687
5	Semana Data	1264.20	41400.0	-40135.80	-96.946377
6	None	25652.04	NaN	NaN	NaN

**Conclusión breve:** Ninguna campaña fue exitosa.  


In [ ]:
# TODO estudiante: completa la consulta SQL de acuerdo con la consigna anterior.
# Sugerencia:
# 1. Identifica la tabla base.
# 2. Define la métrica y la dimensión.
# 3. Valida si necesitas JOIN, WHERE, GROUP BY u ORDER BY.

query = """
SELECT
	ca.campaign_name,
  SUM(pa.amount_paid) as ingreso_neto,
  SUM(ca.spend_usd) as inversion,
  SUM(pa.amount_paid) - SUM(ca.spend_usd) as Ganancia,
  100 * (SUM(pa.amount_paid) - SUM(ca.spend_usd)) / NULLIF(SUM(ca.spend_usd),0) as ROI
FROM orders o
JOIN payments pa
	ON o.order_id = pa.order_id
LEFT JOIN campaigns ca
	ON o.promo_code = ca.promo_code
WHERE o.status <> 'cancelled'
	AND o.status <> 'refunded'
GROUP BY
	ca.campaign_name
ORDER BY
	ROI DESC;
"""

sql(query)


,campaign_name,ingreso_neto,inversion,Ganancia,ROI
0,Reactiva lectores,1222.14,13500.0,-12277.86,-90.947111
1,Cierre semestre,2085.65,37700.0,-35614.35,-94.467772
2,Lecturas de enero,1721.26,34800.0,-33078.74,-95.053851
3,Tech Readers,1244.88,30800.0,-29555.12,-95.958182
4,Impulso Business,1428.60,38400.0,-36971.40,-96.279687
5,Semana Data,1264.20,41400.0,-40135.80,-96.946377
6,None,25652.04,NaN,NaN,NaN


## 12) Validación antes de reportar

Un resultado SQL no se reporta sin controles mínimos.

Checklist de validación:

- ¿Las tablas cargaron con las filas esperadas?
- ¿La llave de unión es correcta?
- ¿El `JOIN` multiplicó filas de forma no intencional?
- ¿El KPI debe excluir órdenes canceladas o reembolsadas?
- ¿La moneda y unidad están claras?
- ¿Las columnas calculadas tienen nombres interpretables?
- ¿El resultado responde una pregunta de negocio?

Plantilla de comunicación:

| Elemento | Ejemplo |
|---|---|
| Hallazgo | La categoría Data concentra el mayor gross profit. |
| Evidencia | Revenue, cost y margin calculados por categoría. |
| Implicación | Puede ser una categoría prioritaria para inventario o campañas. |
| Acción | Revisar disponibilidad y elasticidad de descuentos. |

## Takeaways

- SQL es el lenguaje base para convertir datos tabulares en evidencia analítica.
- En Colab podemos trabajar SQL sin instalar un motor local.
- DuckDB permite consultar tablas en memoria con sintaxis SQL.
- Antes de analizar, se validan tablas, conteos y llaves.
- `SELECT`, `WHERE`, `GROUP BY` y `JOIN` forman el núcleo de una consulta analítica.
- Los KPIs financieros deben tener fórmula, filtro, unidad e interpretación.
- Un buen analista no solo calcula: también comunica supuestos, hallazgos e implicaciones.

## Cierre

En esta sesión pasamos de archivos Parquet en GitHub a consultas SQL en Google Colab.

Flujo final:

```text
GitHub /datasets → Parquet → pandas → DuckDB → SQL → DataFrame → interpretación
```

Para la siguiente sesión, el foco será práctico: resolver preguntas de negocio con SQL, depurar errores y documentar resultados.

### Canales de apoyo

- `DATA-CO-LEARNING`: horarios de atención para dudas puntuales.
- `DA_CONSULTA`: preguntas sobre contenido o proyecto usando el tag correspondiente.
- `SPRINT FOCUS`: sesiones extra para profundizar en temas del sprint.
- `SESIONES 1:1`: tutorías individuales agendadas con antelación.
- Canal de Discord de la cohorte: espacio para compartir recursos, dudas y aprendizajes.